packages/docusaurus-theme-mermaid/src/theme/Mermaid/index.tsx

```diff
@@ -5,28 +5,25 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {useEffect, useState} from 'react';
+import React, {useEffect, useState, useRef} from 'react';
 import useIsBrowser from '@docusaurus/useIsBrowser';
 import mermaid from 'mermaid';
 import type {Props} from '@theme/Mermaid';
 
-/**
- * Assign a unique ID to each mermaid svg as per requirements
- * of `mermaid.render`.
- */
-let id = 0;
-
 export default function Mermaid({value}: Props): JSX.Element {
   // When theme updates, rerender the SVG.
-  const [svg, setSvg] = useState<string>('');
+  const [svg, setSvg] = useState('');
   const isBrowser = useIsBrowser();
+  // Assign a unique ID to each mermaid svg as per requirements of
+  // `mermaid.render`.
+  const id = useRef(0);
 
   useEffect(() => {
     const render = () => {
-      mermaid.render(`mermaid-svg-${id.toString()}`, value, (renderedSvg) =>
```

Comment:
The id should be unique across all mermaid diagrams in the page. i.e. unique across multiple instances of `Mermaid` component. I believe this is because the generated content has a unique id attribute https://github.com/mermaid-js/mermaid/blob/1148039c6ba81e473f4a506fdcb3ac9e6c1a182e/src/mermaidAPI.js#L302